lib/crawl.js

```diff
@@ -81,12 +104,16 @@ function crawl(url, flags) {
     // the correct mimetypes/headers
     crawler.on('fetchcomplete', queueItem => {
         if (queueItem.stateData.contentType.indexOf('text/html') > -1) {
-            urls.push({
-                label: queueItem.path,
-                url: queueItem.url,
-                selectors: ['document'],
-                hideSelectors: ['iframe'],
-            });
+            // Clone the default scenario and adjust urls as needed
+            const currentScenario = {...defaultScenario};
+            currentScenario.label = queueItem.path;
+            if( flags.referenceUrl ){
+                currentScenario.referenceUrl = queueItem.url.replace(url, flags.referenceUrl);
+            } else if( currentScenario.referenceUrl ){
+                delete currentScenario[referenceUrl];
+            }
+            currentScenario.url = queueItem.url;
+            urls.push(currentScenario);
```

Comment:
This could be simplified to something like

```js
urls.push({
    ...defaultScenario,
    label: queueItem.path,
    url: flags.referenceUrl ? queueItem.url.replace(url, flags.referenceUrl) : queueItem.url,
})
```

I'm not sure why you're deleting `currentScenario[referenceUrl];`, why would that be there?

Reply:
@danreeves the `referenceUrl` allows Backstop to test against multiple environments, e.g. a staging and production. Both `url` and `referenceUrl` must be passed for Backstop to compare the two.

I'm deleting `referenceUrl` if it exists in the default config but isn't passed as a flag so that it doesn't have an empty value.

The idea being `referenceUrl` is optional which will allow it's use but maintain backwards compatibility with how `backstop-crawl` functions now.

Reply:
```
if( flags.referenceUrl ){
    currentScenario.referenceUrl = queueItem.url.replace(url, flags.referenceUrl);
} else if( Object.prototype.hasOwnProperty.call(currentScenario, "referenceUrl") ){
    delete currentScenario[referenceUrl];
}
```

Not sure why `delete currentScenario[referenceUrl]` is complaining `referenceUrl is not defined` when it is wrapped in a conditional to check if it exists before deleting.